sirepo/pkcli/runner.py

```diff
@@ -15,31 +15,33 @@
 import functools
 import os
 from pykern.pkdebug import pkdp, pkdc, pkdlog, pkdexc
+from pykern import pkcollections
 from pykern import pkio
 from pykern import pkjson
-import re
-import shlex
-from sirepo import mpi
 from sirepo import runner_client
 from sirepo import srdb
-from sirepo.template import template_common
+from sirepo.runner_daemon import local_process, docker_process
 import subprocess
 import sys
+import time
 import trio
 
-# How often threaded blocking operations should wake up and check for Trio
-# cancellation
-_CANCEL_POLL_INTERVAL = 1
+# Every _TMP_DIR_CLEANUP_TIME seconds, we scan through the run database, and
+# any directories that are named '*.tmp', and whose mtime is
+# >_TMP_DIR_CLEANUP_TIME in the past, are deleted.
+_TMP_DIR_CLEANUP_TIME = 24 * 60 * 60  # 24 hours
 
 _CHUNK_SIZE = 4096
 _LISTEN_QUEUE = 1000
 
 _KILL_TIMEOUT_SECS = 3
-#: Need to remove $OMPI and $PMIX to prevent PMIX ERROR:
-# See https://github.com/radiasoft/sirepo/issues/1323
-# We also remove SIREPO_ and PYKERN vars, because we shouldn't
-# need to pass any of that on, just like runner.docker, doesn't
-_EXEC_ENV_REMOVE = re.compile('^(OMPI_|PMIX_|SIREPO_|PYKERN_)')
+
+_BACKEND_INFO_BASENAME = 'backend.json'
```

Comment:
We save the backend info before starting to run the job, and we write the status after it's done. I suppose we could use the same file and rewrite it, but the lifecycle and consumers are otherwise unrelated, so using two files keeps it decoupled. It doesn't matter too much either way I guess.